### ISSUE TITLE
NET 7 Cli command fail

### ISSUE THREAD

**Issue by DonPangPang** [dotnet/core#8833]
OS: Ubuntu 23
Env: snap dotnet-sdk 7.0.402
project type: blazor-wasm

command: `dotnet workload restore`
install wasm-tools fail...
has no path `/snap/dotnet-sdk/224/metadata`
![image](https://github.com/dotnet/core/assets/49866401/d500f624-b01b-4f2d-9a88-508f7614ddbc)

**Comment by DonPangPang** [comment]
`dotnet workload install wasm-tools` fail, too.

**Comment by liveans** [comment]
Hello,

Looks like your filesystem is read-only. If you have installed dotnet with the `sudo` command then this can happen.
Can you run those commands with `sudo` privilege? Or delete dotnet and install it without `sudo`.

**Comment by DonPangPang** [comment]
`sudo` fail , too.
```
powers@ubuntu:WasmTest$ sudo dotnet workload restore
[sudo] powers 的密码： 

欢迎使用 .NET 7.0!
---------------------
SDK 版本: 7.0.402

遥测
---------
.NET 工具会收集用法数据，帮助我们改善你的体验。它由 Microsoft 收集并与社区共享。你可通过使用喜欢的 shell 将 DOTNET_CLI_TELEMETRY_OPTOUT 环境变量设置为 "1" 或 "true" 来选择退出遥测。

阅读有关 .NET CLI 工具遥测的更多信息: https://aka.ms/dotnet-cli-telemetry

----------------
已安装 ASP.NET Core HTTPS 开发证书。
若要信任该证书，请运行 "dotnet dev-certs https --trust" (仅限 Windows 和 macOS)。
了解 HTTPS: https://aka.ms/dotnet-https
----------------
编写你的第一个应用: https://aka.ms/dotnet-hello-world
查找新增功能: https://aka.ms/dotnet-whats-new
浏览文档: https://aka.ms/dotnet-docs
在 GitHub 上报告问题和查找源: https://github.com/dotnet/core
使用 "dotnet --help" 查看可用命令或访问: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
正在安装工作负载“wasm-tools”
Unhandled exception: System.IO.IOException: Read-only file system : '/snap/dotnet-sdk/224/metadata'
   at System.IO.FileSystem.CreateDirectory(String fullPath, UnixFileMode unixCreateMode)
   at System.IO.Directory.CreateDirectory(String path)
   at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallerFactory.CanWriteToDotnetRoot(String dotnetDir)
   at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallerFactory.GetWorkloadInstaller(IReporter reporter, SdkFeatureBand sdkFeatureBand, IWorkloadResolver workloadResolver, VerbosityOptions verbosity, String userProfileDir, Boolean verifySignatures, INuGetPackageDownloader nugetPackageDownloader, String dotnetDir, String tempDirPath, PackageSourceLocation packageSourceLocation, RestoreActionConfig restoreActionConfig, Boolean elevationRequired)
   at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallCommand..ctor(ParseResult parseResult, IReporter reporter, IWorkloadResolver workloadResolver, IInstaller workloadInstaller, INuGetPackageDownloader nugetPackageDownloader, IWorkloadManifestUpdater workloadManifestUpdater, String dotnetDir, String userProfileDir, String tempDirPath, String version, IReadOnlyCollection`1 workloadIds, String installedFeatureBand)
   at Microsoft.DotNet.Workloads.Workload.Restore.WorkloadRestoreCommand.Execute()
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.DotNet.Cli.Parser.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__17_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass15_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass7_0.<<UseExceptionHandler>b__0>d.MoveNext()

```

and i try to search the path `/snap/dotnet-sdk/224/metadata` fail !
```
powers@ubuntu:WasmTest$ cd /snap/dotnet-sdk/224/
powers@ubuntu:224$ ll
总计 268
drwxr-xr-x 14 root root    281 10月  5 01:45 ./
drwxr-xr-x  4 root root   4096 10月 12 10:04 ../
-rwxr-xr-x  1 root root 181552 10月  5 01:45 dotnet*
drwxr-xr-x  5 root root     52 10月  5 01:45 etc/
drwxrwxr-x  3 root root     26 10月  5 01:45 host/
drwxr-xr-x  4 root root     75 10月  5 01:45 lib/
-rw-rw-r--  1 root root   1116  9月 28 08:55 LICENSE.txt
drwxr-xr-x  2 root root     32 10月  5 01:45 meta/
drwxrwxr-x  6 root root    171 10月  5 01:45 packs/
drwxrwxr-x  3 root root     30 10月  5 01:45 sdk/
drwxrwxr-x  3 root root     30 10月  5 01:45 sdk-manifests/
drwxr-xr-x  4 root root     88 10月  5 01:45 shared/
drwxr-xr-x  2 root root      3 10月  5 01:45 snap/
drwxrwxr-x  3 root root     29 10月  5 01:45 templates/
-rw-rw-r--  1 root root  86284  9月 28 08:55 ThirdPartyNotices.txt
drwxr-xr-x  7 root root    113 10月  5 01:45 usr/
drwxr-xr-x  3 root root     26 10月  5 01:45 var/
powers@ubuntu:224$ 

```

**Comment by DonPangPang** [comment]
> Hello,
> 
> Looks like your filesystem is read-only. If you have installed dotnet with the `sudo` command then this can happen. Can you run those commands with `sudo` privilege? Or delete dotnet and install it without `sudo`.

i use `sudo snap install dotnet-sdk  --classic` to install...

**Comment by DonPangPang** [comment]
it seems that, maybe there is something wrong with the `dotnet path`?
I'm having a similar path issue when using vscode in `C#` and `C# Dev Kit` extensions.
https://github.com/dotnet/vscode-csharp/issues/6513#issuecomment-1758581553

**Comment by liveans** [comment]
@dsplaisted can you help here?

**Comment by dsplaisted** [comment]
@dotnet/dotnet-cli This looks like it may be an issue related to a snap install.  Where do we route that?

**Comment by leecow** [comment]
There are a number of known issues using Snap installs of .NET and we'll need to add workload installation to the mix. In the meantime, the latest 7.0 runtime and sdk are available natively in the 23.04 (lunar) package archives. See for details https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu.

**Comment by DonPangPang** [comment]
> There are a number of known issues using Snap installs of .NET and we'll need to add workload installation to the mix. In the meantime, the latest 7.0 runtime and sdk are available natively in the 23.04 (lunar) package archives. See for details https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu.

okey, thanks.

**Comment by Tychus** [comment]
don't use the snap package. Made that mistake in a fresh install always issues. `apt get install dotnet-sdk-8.0` and all will be good again.
